### PR TITLE
Add diagrams to further enhance Target Allocator documentation - #2229

### PR DESCRIPTION
**Pull request by avillela**
**Description:** Add diagrams to further clarify Target Allocator functionality

**Link to tracking Issue:** #2229 

**Testing:** n/a

**Documentation:** Updated Target Allocator readme and added further clarification to turning on multi-container multi-instrumentation functionality in main readme.

### CODE REVIEW

File: cmd/otel-allocator/README.md [hunk 1]
```diff
@@ -11,11 +11,17 @@ The TA serves two functions:
 
 ## Even Distribution of Prometheus Targets
 
-The Target Allocator's first job is to discover targets to scrape and collectors to allocate targets to. Then it can distribute the targets it discovers among the collectors. This means that the OTel Collectors collect the metrics instead of a Prometheus [scraper](https://uzxmx.github.io/prometheus-scrape-internals.html). Metrics are ingested by the OTel Collectors by way of the [Prometheus Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md).
+The Target Allocator's first job is to discover targets to scrape and collectors to allocate targets to. Then it can distribute the targets it discovers among the collectors. This means that the OTel Collectors collect the metrics instead of a Prometheus [scraper](https://uzxmx.github.io/prometheus-scrape-internals.html). 
+The [Prometheus Receivers](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md) on the Collectors then query the target allocator to ingest Metrics.
```
Comment (secustor) [inline cmd/otel-allocator/README.md:15]:
```suggestion
The [Prometheus Receivers](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md) on the Collectors then query the target allocator for endpoints to scrape metrics from.
```

File: cmd/otel-allocator/README.md [hunk 2]
```diff
@@ -11,11 +11,17 @@ The TA serves two functions:
...
+The Target Allocator also provides for the discovery of [Prometheus Operator CRs](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md), namely the [ServiceMonitor and PodMonitor](https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator#target-allocator). The ServiceMonitor and the PodMonitor don’t do any scraping themselves; their purpose is to inform the Target Allocator (or Prometheus) to add a new job to their scrape configuration. The Target Allocator then adds the job to the OTel Collector’s [Prometheus Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md)’s scrape configuration.
```
Comment (secustor) [inline cmd/otel-allocator/README.md:21]:
```suggestion
The Target Allocator also provides for the discovery of [Prometheus Operator CRs](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md), namely the [ServiceMonitor and PodMonitor](https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator#target-allocator). The ServiceMonitors and the PodMonitors purpose is to inform the Target Allocator (or PrometheusOperator) to add a new job to their scrape configuration. The Target Allocator then provides the jobs to the OTel Collector [Prometheus Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md). 
```

File: cmd/otel-allocator/images/otel_ta_target_distribution.jpeg
Comment (secustor) [inline cmd/otel-allocator/images/otel_ta_target_distribution.jpeg:1]: The collector directly scrapes the metric targets directly. The diagram implies that the target allocator proxies the requests.

Comment (secustor) [reply cmd/otel-allocator/images/otel_ta_target_distribution.jpeg:1]: This is how would do it, tough I would wait for a second opinion on this. 

Merge both graphics into one. 
The `service_discovery` looks good, so you can use it as basis

From `ServiceMonitor` and `PodMonitor` draw arrows to represent that they are referencing the metric targets and from the receivers draw arrows to the targets too. This represents the scrape requests. 
The graphic would also benefit arrow descriptions. 

If this are mermaid graphics, you should also add them to the repository so they can be adapted later on. 

I hope this helps.

Comment (avillela) [reply cmd/otel-allocator/images/otel_ta_target_distribution.jpeg:1]: Thanks @secustor. I also chatted with @jaronoff97 and came up with a revised version of the first diagram. I also put them into Mermaid to make it easier to edit. If these are okay, I can delete the image files.

Comment (secustor) [reply cmd/otel-allocator/images/otel_ta_target_distribution.jpeg:1]: LGTM

Comment (avillela) [reply cmd/otel-allocator/images/otel_ta_target_distribution.jpeg:1]: Removed the images in favor of the mermaid diagrams.

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**